src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -83,6 +89,116 @@ async function autofillDefaultTransactionValues(
   }
 }
 
+function createSimulatedTransaction(
+  blockchain: Blockchain,
+  options: EthereumInternalOptions,
+  common: Common,
+  transaction: any,
+  simulationBlock: Block
+): SimulationTransaction {
+  const simulationBlockHeader = simulationBlock.header;
+
+  let gas: Quantity;
+  if (typeof transaction.gasLimit === "undefined") {
+    if (typeof transaction.gas !== "undefined") {
+      gas = Quantity.from(transaction.gas);
+    } else {
+      // eth_call isn't subject to regular transaction gas limits by default
+      gas = options.miner.callGasLimit;
+    }
+  } else {
+    gas = Quantity.from(transaction.gasLimit);
+  }
+
+  let data: Data;
+  if (typeof transaction.data === "undefined") {
+    if (typeof transaction.input !== "undefined") {
+      data = Data.from(transaction.input);
+    }
+  } else {
+    data = Data.from(transaction.data);
+  }
+
+  // eth_call doesn't validate that the transaction has a sufficient
+  // "effectiveGasPrice". however, if `maxPriorityFeePerGas` or
+  // `maxFeePerGas` values are set, the baseFeePerGas is used to calculate
+  // the effectiveGasPrice, which is used to calculate tx costs/refunds.
+  const baseFeePerGasBigInt = simulationBlockHeader.baseFeePerGas
+    ? simulationBlockHeader.baseFeePerGas.toBigInt()
+    : undefined;
+
+  let gasPrice: Quantity;
+  const hasGasPrice = typeof transaction.gasPrice !== "undefined";
+  // if the original block didn't have a `baseFeePerGas` (baseFeePerGasBigInt
+  // is undefined) then EIP-1559 was not active on that block and we can't use
+  // type 2 fee values (as they rely on the baseFee)
+  if (!common.isActivatedEIP(1559) || baseFeePerGasBigInt === undefined) {
+    gasPrice = Quantity.from(hasGasPrice ? 0 : transaction.gasPrice);
+  } else {
+    const hasMaxFeePerGas = typeof transaction.maxFeePerGas !== "undefined";
+    const hasMaxPriorityFeePerGas =
+      typeof transaction.maxPriorityFeePerGas !== "undefined";
+
+    if (hasGasPrice && (hasMaxFeePerGas || hasMaxPriorityFeePerGas)) {
+      throw new Error(
+        "both gasPrice and (maxFeePerGas or maxPriorityFeePerGas) specified"
+      );
+    }
+    // User specified 1559 gas fields (or none), use those
+    let maxFeePerGas = 0n;
+    let maxPriorityFeePerGas = 0n;
+    if (hasMaxFeePerGas) {
+      maxFeePerGas = BigInt(transaction.maxFeePerGas);
+    }
+    if (hasMaxPriorityFeePerGas) {
+      maxPriorityFeePerGas = BigInt(transaction.maxPriorityFeePerGas);
+    }
+    if (maxPriorityFeePerGas > 0 || maxFeePerGas > 0) {
+      const a = maxFeePerGas - baseFeePerGasBigInt;
+      const tip = a < maxPriorityFeePerGas ? a : maxPriorityFeePerGas;
+      gasPrice = Quantity.from(baseFeePerGasBigInt + tip);
+    } else {
+      gasPrice = Quantity.from(0);
+    }
+  }
+
+  // if EIP 2930 is activated and they provided an invalid access list,
```

Comment:
Yes, I think you're right. I started down this route because of your comment, then realized it'll be a bit of work to do properly. Our regular transactions with access lists (not the simulated ones) also need better validation, so I've made issue #3377 for this and added a TODO.